Travis said it..
Interesting definition of "healing" you've got.

``` bash
$ build_marlin
Picked up JAVA_TOOL_OPTIONS: 
Loading configuration...
Xlib:  extension "RANDR" missing on display ":1.0".
Initializing packages...
Preparing boards...
Verifying...
In file included from sketch/Marlin.h:382:0,
                 from sketch/M100_Free_Mem_Chk.cpp:43:
sketch/buzzer.h: In member function 'void Buzzer::tone(const uint16_t&, const uint16_t&)':
buzzer.h:109: error: 'thermalManager' was not declared in this scope
         thermalManager.manage_heater();
         ^
exit status 1
'thermalManager' was not declared in this scope
```

--- a/Marlin/buzzer.h
+++ b/Marlin/buzzer.h
@@ -24,8 +24,8 @@
 #define __BUZZER_H__
 
 #include "fastio.h"
-#include "watchdog.h"
 #include "circularqueue.h"
+#include "temperature.h"
 
 #define TONE_QUEUE_LENGTH 4
 
@@ -106,9 +106,7 @@ class Buzzer {
       while (buffer.isFull()) {
         delay(5);
         this->tick();
-        #if ENABLED(USE_WATCHDOG)
-          watchdog_reset();
-        #endif
+        thermalManager.manage_heater();
       }
       this->buffer.enqueue((tone_t) { duration, frequency });
     }

